File: IntentJavaTestCase/app/src/main/java/com/example/intentjavatestcase/MainActivity.java
package com.example.intentjavatestcase;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    Button senddata,senddata1;
    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textval);
        senddata = findViewById(R.id.senddata);
        senddata1 = findViewById(R.id.senddata1);

        senddata.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, SecondActivity.class);
                i.putExtra("name","Ashish Ranjan");   //for method 1 and 2 comment this line
                startActivity(i);
            }
        });

        senddata1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                textView.setVisibility(View.VISIBLE);
            }
        });
    }
}

File: IntentkotlinTestCase/app/src/main/java/com/example/intentkotlintestcase/MainActivity.kt
package com.example.intentkotlintestcase

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var sendbutton1: Button =  findViewById(R.id.sendbutton1);
        var sendbutton2: Button =  findViewById(R.id.sendbutton2);
        sendbutton1.setOnClickListener(View.OnClickListener
        {
            val intent= Intent(this,SecondActivity::class.java)
            intent.putExtra("name","Ranjan")    //for method 3 else comment this
            startActivity(intent)
        })

        val textval:TextView = findViewById(R.id.textval)
        sendbutton2.setOnClickListener(View.OnClickListener {
            textval.visibility = View.VISIBLE
        })


    }
}